src/Abuse/Adapters/TimeLimit.php

```diff
@@ -207,45 +202,41 @@ protected function hit(string $key, int $time): void
      *
      * Returns logs with an offset and limit
      *
-     * @param $offset 
+     * @param $offset
      * @param $limit
-     * 
+     *
      * @return array
      */
-    public function getLogs(int $offset, int $limit): array {  
-
-        $st = $this->getPDO()->prepare('SELECT * FROM `' . $this->getNamespace() . '.abuse.abuse`;
-            LIMIT :offset, :limit
-        ');
-        $st->bindValue(':offset',     $offset,    PDO::PARAM_INT);
-        $st->bindValue(':limit',     $limit,    PDO::PARAM_INT);
-        $st->execute();
-        
-        $result = $st->fetchAll();
-        
+    public function getLogs(int $offset, int $limit): array
+    {
+        Authorization::disable();
+        $result = $this->db->find(TimeLimit::COLLECTION, [], $limit, $offset);
+        Authorization::reset();
         return $result;
     }
 
-
     /**
      * Delete logs older than $timestamp seconds
-     * 
-     * @param int $timestamp 
-     * 
-     * @return bool   
+     *
+     * @param int $timestamp
+     *
+     * @return bool
      */
-    public function cleanup(int $timestamp):bool
+    public function cleanup(int $timestamp): bool
     {
-        $st = $this->getPDO()->prepare('DELETE 
-        FROM `'.$this->getNamespace().'.abuse.abuse`
-            WHERE `_time` < :timestamp');
+        Authorization::disable();
+        $documents = $this->db->find(TimeLimit::COLLECTION, [
```

Comment:
Will this actually clean up all necessary documents? `\Utopia\Database\Database->find()` has a default limit of 25 records:
https://github.com/utopia-php/database/blob/main/src/Database/Database.php#L813

Reply:
Good catch, we should iterate the results until no results come up - similar to what we do on the deletes worker.